pkg/build/builder/common.go

```diff
@@ -62,7 +62,10 @@ func buildInfo(build *api.Build) []KeyValue {
 		if build.Spec.Source.Git.Ref != "" {
 			kv = append(kv, KeyValue{"OPENSHIFT_BUILD_REFERENCE", build.Spec.Source.Git.Ref})
 		}
-		if build.Spec.Revision != nil && build.Spec.Revision.Git != nil && build.Spec.Revision.Git.Commit != "" {
+
+		if sourceInfo != nil {
+			kv = append(kv, KeyValue{"OPENSHIFT_BUILD_COMMIT", sourceInfo.CommitID})
```

Comment:
Is it possible that sourceInfo.CommitID could be blank? If so, then it'd be good to check for it.